### PR TITLE
Add pylint workflow

### PR DESCRIPTION
**Pull request by bveeramani**
This PR implements a GitHub Action that runs `pylint` when PRs are opened or updated.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -27,7 +27,6 @@
 
 test_requirements = [
     'pytest==6.1.1',
-    'pytest-pylint==0.17.0',
```
Comment (nums-project) [inline setup.py:30]: Can we re-add this as a requirement for testing? We sometimes want to be able to run linting locally without having to install the required files.

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: lint
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
```
Comment (nums-project) [inline .github/workflows/lint.yml:5]: Could you briefly describe "synchronize?"

Comment (bveeramani) [reply .github/workflows/lint.yml:5]: Synchronize refers to when you commit changes to an existing PR.

File: setup.py (hunk 1 above)
Comment (bveeramani) [reply setup.py]: Fixed

**Comment by nums-project** [comment]
Is there a way to test that linting is working properly? Is there a way to observe the output of the linting check?

**Pull request merged**